Add support pasting key value entries update ArrayList pasting

codyrancher opened the pull request:
KeyValue:
    - If pasted with only one line and doesn't have a ':' or '=' token it will perform a normal paste.
    - If pasted in the key column it will replace the current row.
    - Performs a normal paste in the value field
    
 ArrayList:
    - Default to single line
    - Only handles pasting when it's single line
    - Only shows the default protip if it's single line
    
 rancher/dashboard#1701

vincent99 reviewed (changes requested):
Paste into the `value` should just do regular native paste into that one field.  There are multiple common use cases this will break as-is, like any configmap or secrets that you want to paste JSON or YAML into. (This is also why the protip only mentions pasting into the `key` field; keys aren't multiline.)

ArrayList is where you get into the "is it 3 lines for one value or 3 values that should become 3 rows" problem, which KeyValue avoids by pasting into the single-lined key...
- If `valueMultiline=true`, do native paste and hide the default protip.
- Default `valueMultiline` prop to false
- Evaluate existing usages of ArrayList and set `:value-multiline="true"` on the ones that have a reason to need it.  Many/most are single word-type values like an IP/hostname, or use the slot to show a `<select>`.
- _Can_ you paste anything if it's a select?  Probably the tip shouldn't be shown here either.

codyrancher commented:
I addressed your feedback.

While looking through ArrayList it seems we never make use of multiline. Multiline didn't fully function because of trimming that was being done. Maybe we can consider removing multiline? 

- I fixed the multiline trim issue.
- I made sure all of the ArrayLists that customize the column slot now hide the protip. 
- I think there are a few places we could enable the protip but I didn't do that with this pass.
